roles/nv_gpu_install_from_commit/tasks/build.yml

```diff
@@ -67,33 +69,42 @@
 - name: delete operator image builder pod, if any
   shell: oc delete -f "{{ gpu_operator_image_builder_pod }}" || true
 
-- name: delete operator image, if any
-  shell: oc delete imagestreamtag/gpu-operator-ci:gpu-operator -n gpu-operator-ci || true
+- name: prepare the operator image
+  block:
+  - name: check if the operator image exists
+    shell: "oc get imagestreamtag/gpu-operator-ci:{{ gpu_operator_image_tag }} -n gpu-operator-ci"
 
-- name: apply operator image builder pod manifest
-  shell: sed 's|{{ '{{' }} builder_secret {{ '}}' }}|{{ builder_secret.stdout }}|' "{{ gpu_operator_image_builder_pod }}" \
-       | sed 's|{{ '{{' }} gpu_operator_git_repo {{ '}}' }}|{{ gpu_operator_git_repo }}|' \
-       | sed 's|{{ '{{' }} gpu_operator_git_ref {{ '}}' }}|{{ gpu_operator_git_ref }}|' \
-       | sed 's|{{ '{{' }} gpu_operator_image_tag {{ '}}' }}|{{ gpu_operator_image_tag }}|' \
-       | oc apply -f-
-  args:
-    warn: false # don't warn about using sed here
+  rescue:
+  - debug:
+      msg: the operator image does not exist, build it
+  - name: apply operator image builder pod manifest
+    shell: sed 's|{{ '{{' }} builder_secret {{ '}}' }}|{{ builder_secret.stdout }}|' "{{ gpu_operator_image_builder_pod }}" \
+         | sed 's|{{ '{{' }} gpu_operator_git_repo {{ '}}' }}|{{ gpu_operator_git_repo }}|' \
+         | sed 's|{{ '{{' }} gpu_operator_git_ref {{ '}}' }}|{{ gpu_operator_git_ref }}|' \
+         | sed 's|{{ '{{' }} gpu_operator_image_tag {{ '}}' }}|{{ gpu_operator_image_tag }}|' \
+         | oc apply -f-
+    args:
+      warn: false # don't warn about using sed here
 
-- name: wait for the operator image to be built
-  block:
   - name: wait for the operator image to be built
-    shell: oc get pod --field-selector=status.phase=Succeeded,metadata.name=operator-image-builder-pod --no-headers -oname -n gpu-operator-ci | grep operator-image-builder-pod
-    register: wait_image_builder_pod
-    until: wait_image_builder_pod.rc != 1
-    retries: 40
-    delay: 30
-  rescue:
-  - name: get info about operator image build failure
-    command: oc get pods -n gpu-operator-ci
-    ignore_errors: true
-  - name: get logs of image build failure
-    command: oc get logs pod/operator-image-builder-pod
-    ignore_errors: true
-  - name: operator image failed to build
-    fail:
-      msg: operator image failed to build
+    block:
+    - name: wait for the operator image to be built
+      shell: oc get pod --field-selector=status.phase=Succeeded,metadata.name=operator-image-builder-pod \
```

Comment:
If it is going to be a multiliner I prefer to do `shell: -|` and put everything in the next line

Reply:
```
      shell: -|
        echo toto
        truc
```
is not valid: `"cmd": "-| echo toto truc"`
but this one works as expected, I'll switch to it for "multi-line one-liners" :
```
      shell: >-
        echo toto
        truc
```

Reply:
hum, weird,

```
      shell: >-
        echo toto
        truc
```
^^ this works as expected

```
      shell: >-
        echo toto
            truc
```
^^ this **doesn't** work!

https://stackoverflow.com/questions/54769350/how-to-fold-multiline-yaml-string-with-identations


```
      shell:
        echo toto
            truc
```
so ^^ this is the solution -,-